mapper/cumulocity/c8y_translator_lib/src/serializer.rs

```diff
@@ -126,3 +126,171 @@ impl GroupedMeasurementVisitor for C8yJsonSerializer {
+            "temperature":{
+                "temperature":{
+                    "value": 25.5
+                }
+            },
+             "location": {
+                 "alti": {
+                     "value": 2100.4
+                 },
+                 "longi":{
+                     "value": 2200.4
+                 },
+                 "lati":{
+                     "value": 2300.4
+                 },
+             },
+             "pressure":{
+                 "pressure":{
+                     "value":255.2
+                 }
+             }
+
+        });
+
+        assert_json_eq!(
+            serde_json::from_slice::<serde_json::Value>(&output).unwrap(),
+            expected_output
+        );
+    }
+
+    #[test]
+    fn serialize_empty_message() {
+        let serializer = C8yJsonSerializer::new().unwrap();
+        let expected_output: Vec<u8> = format!(r#"{{"type": "ThinEdgeMeasurement"}}"#).into_bytes();
+        let output = serializer.bytes().unwrap();
+
+        assert_eq!(expected_output.to_vec(), output);
+    }
+
+    #[test]
+    fn serialize_timestamp_message() {
+        let mut serializer = C8yJsonSerializer::new().unwrap();
+        let timestamp = test_timestamp();
+        serializer.timestamp(timestamp).unwrap();
+        let expected_output: Vec<u8> = format!(
+            r#"{{"type": "ThinEdgeMeasurement","time":"{}"}}"#,
+            timestamp.to_rfc3339()
+        )
+        .into();
+        let output = serializer.bytes().unwrap();
+        assert_eq!(expected_output, output);
+    }
+
+    #[test]
+    fn serialize_timestamp_within_group() {
+        let mut serializer = C8yJsonSerializer::new().unwrap();
+        let timestamp = test_timestamp();
+        serializer.start_group("location").unwrap();
+        let result = serializer.timestamp(timestamp);
+
+        assert_matches!(
+            result.unwrap_err(),
+            C8yJsonSerializationError::MeasurementCollectorError(
+                MeasurementStreamError::UnexpectedTimestamp
+            )
+        );
+    }
+
+    #[test]
+    fn serialize_unexpected_end_of_group() {
+        let mut serializer = C8yJsonSerializer::new().unwrap();
+        serializer.measurement("alti", 2100.4).unwrap();
+        serializer.measurement("longi", 2200.4).unwrap();
+        let result = serializer.end_group();
+
+        assert_matches!(
+            result.unwrap_err(),
+            C8yJsonSerializationError::MeasurementCollectorError(
+                MeasurementStreamError::UnexpectedEndOfGroup
+            )
+        );
+    }
+
+    #[test]
+    fn serialize_unexpected_start_of_group() {
+        let mut serializer = C8yJsonSerializer::new().unwrap();
+        serializer.start_group("location").unwrap();
+        serializer.measurement("alti", 2100.4).unwrap();
+        serializer.measurement("longi", 2200.4).unwrap();
+        let result = serializer.start_group("location2");
+
+        assert_matches!(
+            result.unwrap_err(),
```

Comment:
instead of `unwrap_err`, you can simple match against `Err(...)`

Reply:
fixed